packages/story-editor/src/components/panels/shared/hotlinkModal.js

```diff
@@ -0,0 +1,205 @@
+  if (allowedFileTypes.length) {
+    description = sprintf(
+      /* translators: %s is a list of allowed file extensions. */
+      __('You can insert %s.', 'web-stories'),
+      translateToExclusiveList(allowedFileTypes)
+    );
+  }
+
+  useLayoutEffect(() => {
+    // Wait one tick to ensure the input has been loaded.
+    const timeout = setTimeout(() => {
+      if (isOpen && inputRef.current) {
+        inputRef.current.focus();
+      }
+    });
+    return () => clearTimeout(timeout);
+  }, [isOpen, inputRef]);
+
+  const onBlur = useCallback(() => {
+    if (link?.length > 0) {
+      const newLink = withProtocol(link);
+      setLink(newLink);
+      if (!isValidUrlForHotlinking(newLink)) {
+        setErrorMsg(__('Invalid link.', 'web-stories'));
+      }
+    }
+  }, [link]);
+
+  const onChange = useCallback(
+    (value) => {
+      // Always set the error to null when changing.
+      if (errorMsg) {
+        setErrorMsg(null);
+      }
+      setLink(value);
+    },
+    [setLink, errorMsg]
+  );
+
+  const onInsert = useCallback(async () => {
+    if (!link) {
+      return;
+    }
+
+    if (!isValidUrlForHotlinking(link)) {
+      setErrorMsg(__('Invalid link.', 'web-stories'));
+      return;
+    }
+
+    setIsInserting(true);
+
+    try {
+      const hotlinkInfo = await getHotlinkInfo(link);
+      const shouldProxy = await checkResourceAccess(link);
+
+      // After getting link metadata and before actual insertion
+      // is a great opportunity to measure usage in a reasonably accurate way.
+      trackEvent('hotlink_file', {
+        event_label: link,
+        file_size: hotlinkInfo.fileSize,
+        file_type: hotlinkInfo.mimeType,
+        needs_proxy: shouldProxy,
+      });
+
+      onSelect({ src: link, needsProxy: shouldProxy });
+    } catch (err) {
+      trackError('hotlink_file', err?.message);
+
+      setErrorMsg(getErrorMessage(err.code, description));
+    } finally {
+      setIsInserting(false);
+    }
+  }, [
+    description,
+    onSelect,
+    link,
+    getHotlinkInfo,
+    setErrorMsg,
+    checkResourceAccess,
+  ]);
+
+  const onSubmit = useCallback(
+    (evt) => {
+      evt.preventDefault();
+
+      if (!isDisabled) {
+        onInsert();
+      }
+    },
+    [isDisabled, onInsert]
+  );
+
+  return (
+    <Dialog
+      onClose={() => {
+        onClose();
+        setLink('');
+        setErrorMsg(false);
+        setIsInserting(false);
+      }}
+      isOpen={isOpen}
+      title={title}
+      onPrimary={() => onInsert()}
+      primaryText={primaryText}
+      secondaryText={__('Cancel', 'web-stories')}
+      primaryRest={{ disabled: isDisabled }}
+    >
+      <InputWrapper onSubmit={onSubmit}>
+        <Input
+          ref={inputRef}
+          onChange={({ target: { value } }) => onChange(value)}
+          value={link}
+          hint={errorMsg?.length ? errorMsg : description}
+          hasError={Boolean(errorMsg?.length)}
```

Comment:
Given this happens with the input maybe outside the scope of this PR but when using VoiceOver the error isn't announced.

Should there maybe be a standalone error with a live region to announce the input error for this use case ?

Reply:
Good call!

I created #11150 the other day to improve our usage of aria-live regions. Will add these types of errors as an example there.